cdap-cli/src/main/java/co/cask/cdap/cli/CLIConfig.java

```diff
@@ -117,7 +141,8 @@ public void setConnectionConfig(@Nullable ConnectionConfig connectionConfig) {
   public void tryConnect(ConnectionConfig connectionConfig, PrintStream output, boolean debug) throws Exception {
     try {
       AccessToken accessToken = acquireAccessToken(clientConfig, connectionConfig, output, debug);
-      checkConnection(clientConfig, connectionConfig, accessToken);
+      connectionConfig = clientConfig.getConnectionConfig();
```

Comment:
This introduces a bug, where you can't connect to any cluster other than the default localhost one. The `connectionConfig` arg is intended to be the new connection, whereas `clientConfig.getConnectionConfig()` is the current connection. Also, try to avoid modifying method args.